scripts/parse_nagim.py

```diff
@@ -0,0 +1,467 @@
+
+    # Parse KCCG metadata (two columns: sample and project)
+    proj_by_nagim_id = dict()
+    with open(sample_to_project_path) as f:
+        for line in f:
+            if line.strip():
+                nagim_id, proj = line.strip().split('\t')
+                nagim_id = nagim_id.split('.')[
+                    0
+                ]  # AAACD__ST-E00141_HKJ2MCCXX.3 -> AAACD__ST-E00141_HKJ2MCCXX
+                proj_by_nagim_id[nagim_id] = PROJECT_ID_MAP[proj]
+
+    # Setting up project IDs
+    for sample in samples_from_found_files:
+        if sample.nagim_id not in proj_by_nagim_id:
+            logger.critical(
+                f'Sample {sample.nagim_id} not found in {sample_to_project_path}'
+            )
+            sys.exit(1)
+
+        sample.project_id = proj_by_nagim_id[sample.nagim_id]
+
+    # Fixing sample IDs. Some samples (tob-wgs and acute-care)
+    # have CPG IDs as nagim ids, some don't
+    for sample in samples_from_found_files:
+        if sample.nagim_id in extid_to_cpgid:
+            sample.ext_id = sample.nagim_id
+            sample.cpg_id = extid_to_cpgid[sample.nagim_id]
+        elif sample.nagim_id in cpgid_to_extid:
+            sample.ext_id = cpgid_to_extid[sample.nagim_id]
+            sample.cpg_id = sample.nagim_id
+        else:
+            sample.ext_id = sample.nagim_id
+
+    return samples_from_found_files
+
+
+def _transfer_cram_from_terra(upload_bucket: str, prod: bool, hbatch=None):
+    for bucket in SRC_BUCKETS_TEST if not prod else SRC_BUCKETS_MAIN:
+        # Find all crams and crais and copy them in parallel
+        cmd = (
+            f'gsutil ls '
+            f"'{bucket}/WholeGenomeReprocessingMultiple/*/call-WholeGenomeReprocessing/shard-*/WholeGenomeReprocessing/*/call-WholeGenomeGermlineSingleSample/WholeGenomeGermlineSingleSample/*/"
+            f"call-BamToCram/BamToCram/*/call-ConvertToCram/**/*.cram*'"
+            f' | gsutil -m cp -I {upload_bucket}/'
+        )
+        if hbatch:
+            _add_batch_job(hbatch, cmd, f'Transfer CRAMs from {bucket}')
+        else:
+            _call(cmd)
+
+
+def _transfer_gvcf_from_terra(upload_bucket: str, prod: bool, hbatch=None):
+    for bucket in SRC_BUCKETS_TEST if not prod else SRC_BUCKETS_MAIN:
+        # Find all GVCFs and TBIs and copy them in parallel
+        cmd = (
+            f'gsutil ls '
+            f"'{bucket}/WholeGenomeReprocessingMultiple/*/call-WholeGenomeReprocessing/shard-*/WholeGenomeReprocessing/*/call-WholeGenomeGermlineSingleSample/WholeGenomeGermlineSingleSample/*/"
+            f"call-BamToGvcf/VariantCalling/*/call-MergeVCFs/**/*.hard-filtered.g.vcf.gz*'"
+            f' | gsutil -m cp -I {upload_bucket}/'
+        )
+        if hbatch:
+            _add_batch_job(hbatch, cmd, f'Transfer GVCFs from {bucket}')
+        else:
+            _call(cmd)
+
+
+def _add_batch_job(hbatch, cmd, job_name):
+    j = hbatch.new_job(job_name)
```

Comment:
Why default 32 cores?

Reply:
The `-m` option for gsutil enables multi-threaded copy, I think the more cores the faster?